### PR TITLE
The Edit Task Command

### PR DESCRIPTION
**Pull request by Yskie**
User can now edit task based on the index of the `index` in the task list. The command for editing tasks is:
 ```
edittask INDEX [n/TASK_NAME] [d/TASK_DESCRIPTION] [p/TASK_PRIORITY] [by/TASK_DEADLINE] 
```
Parameter: `INDEX`
- Must be a numeric string representing a number between 1 and number of tasks.

Optional Parameters:
`TASK_NAME`
`TASK_DESCRIPTION`
`TASK_PRIORITY`
`TASK_DEADLINE`

## Things to note
At least one of the optional fields must be present for the command to edit the tasks.
 
This PR is linked to #45.

### CODE REVIEW

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -183,8 +192,7 @@ public TaskList getTaskList() {
     }
 
     /**
-     * Returns an unmodifiable view of the list of {@code Task} backed by the internal list of
-     * {@code versionedAddressBook}
+     * Returns an unmodifiable view of the list of {@code Task}.
```
Comment (breezetall) [inline src/main/java/seedu/address/model/ModelManager.java:195]: Should this say modifiable instead?

File: src/main/java/seedu/address/model/TaskList.java [hunk 2]
```diff
@@ -80,30 +79,34 @@ public boolean hasTask(Task task) {
...
-                                return 1;
-                            }
-                            return 0;
-                        }
-                    } else {
-                        return o2.getPriority().getValue() - o1.getPriority().getValue();
-                    }
-                } else {
+
+                // Sort by status
```
Comment (breezetall) [inline src/main/java/seedu/address/model/TaskList.java:88]: Looks good! Thank you!

**Review by breezetall — APPROVED** [review]
LGTM! Just a minor javadoc issue.

**Review by chin-herng — APPROVED** [review]
LGTM LFG!

**Review by Kaya3842 — APPROVED** [review]
LGTM!

**Pull request merged**